althea_kernel_interface/src/ip_route.rs

```diff
@@ -12,47 +28,24 @@ impl KernelInterface {
 
         let stdout = String::from_utf8(output.stdout).unwrap();
 
-        let mut def_route = Vec::new();
-
         // find all lines
-        for i in stdout.lines() {
-            // starting with default
-            if i.starts_with("default") {
-                for j in i.split_whitespace() {
-                    if j.len() != 0 {
-                        def_route.push(String::from(j));
-                    }
-                }
-                return Some(def_route);
-            }
+        for i in stdout.lines().filter(|line| line.starts_with("default")) {
+            return Some(i.split_whitespace().map(|s| s.to_string()).collect());
```

Comment:
could we get a commit about what exactly this operation achieves? It's kinda hard to unpack

Reply:
Before we iterate over tokens and create another vector to move tokens that aren't empty. Now it leverages the fact that `split_whitespace` filters out excess whitespaces (and empty tokens). There is some functional magic there to convert iterable of &str into iterable of `String` (`s.to_string()`) and `collect` creates a `Vec<String>`.

I'll add some comments.